src/NHibernate/Type/UriType.cs

```diff
@@ -84,9 +84,7 @@ public override object FromStringValue(string xml)
 		}
 
 		public string ObjectToSQLString(object value, Dialect.Dialect dialect)
-		{
-			return "'" + ((Uri)value).OriginalString + "'";
-		}
+			=> dialect.ToStringLiteral(((Uri) value).OriginalString, SqlType);
```

Comment:
I have to check if `Uri` may accept an original value allowing SQL injections (very likely yes), and if yes, add another test case.